src/Pipes/GZip.hs

```diff
@@ -62,8 +62,8 @@ decompress' = Pipes.Zlib.decompress' gzWindowBits
 compress
   :: MonadIO m
   => Pipes.Zlib.CompressionLevel
-  -> Proxy x' x () B.ByteString m r -- ^ Decompressed stream
-  -> Proxy x' x () B.ByteString m r -- ^ Compressed stream
+  -> Producer B.ByteString m r -- ^ Decompressed stream
+  -> Producer' B.ByteString m r -- ^ Compressed stream
```

Comment:
Yes, this is the type that GHC gives me if I remove the type signature.

However I think that you're mistaken (though I could be wrong, thinking about this makes my head hurt...).  This signature will still accept a fully polymorphic producer!  

Indeed, in any examples that I run, that is exactly what it is doing eg the producer created by a `BS.fromHandle` is polymorphic and is accepted here.